src/peer_manager.rs

```diff
@@ -401,6 +426,17 @@ impl PeerManager {
             merge_prefix: merge_prefix,
             groups: groups_as_names,
         };
+        let mut expected_peers = HashMap::new();
+        mem::swap(&mut self.expected_peers, &mut expected_peers);
+        expected_peers.extend(own_merge_details.groups
+            .values()
+            .flat_map(|group| group.iter())
+            .filter_map(|name| if self.peer_map.get_by_name(name).is_none() {
```

Comment:
Since we only remove from `expected_peers` in `add_to_routing_table`, maybe the right criterion here would be whether it's already in `Routing` state. Every peer that isn't probably should go in?